### ISSUE TITLE
Karabo supports all supported Python versions

### ISSUE THREAD

**Issue by Lukas113** [i4Ds/Karabo-Pipeline#519]
Karabo should support all Python versions which are in the current [lifecycle](https://devguide.python.org/versions/).
TODO:
- [Karabo Feedstock](https://github.com/i4Ds/Karabo-Feedstock/tree/main/.github/workflows) need builds for each version
- Karabo code should be compatible with all versions (if incompatible code exists, if-else with python-verisons, just use oldest python-typing features)
- Test setup for each python version
- Loose dependencies accordingly (tests should tell us the versions)

Due to resource constraints, this will likely be delayed by several months.

**Comment by sfiruch** [comment]
According to #225 the feedstock should be 3.10 compatible.

**Comment by kenfus** [comment]
This would then also allow us to easily add an example notebook on google colab, which uses python 3.10 in the background.

**Comment by Lukas113** [comment]
> This would then also allow us to easily add an example notebook on google colab, which uses python 3.10 in the background.

I don't think that this has an influence whether Karabo runs on colab or not, because Karabo needs conda anyway. And with conda, you can install any python-version.

**Comment by kenfus** [comment]
> > This would then also allow us to easily add an example notebook on google colab, which uses python 3.10 in the background.
> 
> I don't think that this has an influence whether Karabo runs on colab or not, because Karabo needs conda anyway. And with conda, you can install any python-version.

Sadly, no. It somehow links to the runtime python, it's not independent. The `condacolab` package is quite hacky, but it works.

**Comment by Lukas113** [comment]
Ok, that's very surprising, because the advantage of conda lies in installing non-python deps like Python. And loosing such an advantage and still allowing conda doesn't make any sense to me.

**Comment by anawas** [comment]
Set this to priority high, because we're still on version 3.9. Its expected end-of-life is however not before 10.2025. But versions 3.10 and 3.11 are widely used.

**Comment by d3v-null** [comment]
With the Python3.9 end of life coming in 5 months, I had a quick look for things preventing a python upgrade, and it seems like the way that wsclean or everybeam is currently configured in Feedstock might be incompatible with 3.10 https://github.com/i4Ds/Karabo-Feedstock/issues/28
